src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -421,7 +428,7 @@ protected function build(ContainerBuilder $container)
      */
     protected function getContainerClass()
     {
-        return $this->name.ucfirst($this->environment).($this->debug ? 'Debug' : '').'ProjectContainer';
+        return 'app'.str_replace('\\', '_', \get_class($this)).ucfirst($this->environment).($this->debug ? 'Debug' : '').'ProjectContainer';
```

Comment:
the 'app' prefix can be removed as the class must be valid due to the kernel class (not starting with numbers for example). also I hoped you would name it Container instead of ProjectContainer

Reply:
done/done